pkcs11/src/new/types/mechanism/rsa.rs

```diff
@@ -133,3 +135,25 @@ impl From<PkcsOaepParams> for Mechanism {
         Mechanism::RsaPkcsOaep(pkcs_oaep_params)
     }
 }
+
+#[cfg(feature = "psa-crypto-conversions")]
+#[allow(deprecated)]
+impl TryFrom<psa_crypto::types::algorithm::Hash> for PkcsMgfType {
```

Comment:
I'm not convinced this is entirely correct to do - Message Generation Functions aren't plain hash functions: 
> CK_RSA_PKCS_MGF_TYPE  is used to indicate the Message Generation Function (MGF) applied to a message block when formatting a message block for the PKCS #1 OAEP encryption scheme or the PKCS #1 PSS signature scheme.

It does encode things differently per hash type, but maybe it's better to have a method on `PkcsMgfType` to convert rather than a generic `TryFrom`

Reply:
As in you mean there could be other algorithms than just hashes? I can create a method `from_hash` but in the end isn't it similar?

Reply:
No, as in - MGFs are not hash functions (though they are parametrized with hash functions), having a conversion between them sort of implies that they're the same thing just expressed in different ways.